core-rust/mesh-api-server/src/mesh_api/conversions/block.rs

```diff
@@ -13,8 +13,49 @@ pub(crate) fn to_mesh_api_block_identifier_from_state_version(
     })
 }
 
-pub(crate) fn to_mesh_api_block_identifier_from_ledger_header(
+pub fn to_mesh_api_block_identifier_from_ledger_header(
     ledger_header: &LedgerStateSummary,
 ) -> Result<models::BlockIdentifier, MappingError> {
     to_mesh_api_block_identifier_from_state_version(ledger_header.state_version)
 }
+
+pub fn extract_state_version_from_mesh_api_partial_block_identifier(
+    block_identifier: &models::PartialBlockIdentifier,
+) -> Result<Option<StateVersion>, ExtractionError> {
+    let state_version = if let Some(index) = block_identifier.index {
+        Some(StateVersion::of(index as u64))
+    } else if let Some(hash) = &block_identifier.hash {
+        let index = hash
+            .parse::<i64>()
+            .map_err(|_| ExtractionError::InvalidInteger {
+                message: "Error converting hash to integer".to_string(),
+            })?;
+        Some(StateVersion::of(index as u64))
```

Comment:
MINOR: If both are provided, we should probably check they agree, like in `extract_state_version_from_mesh_api_block_identifier`

Reply:
Good point

Reply:
Done